### PR TITLE
highlight king when in check

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: chess/src/board_state.rs [hunk 1]
```diff
@@ -8,13 +8,28 @@ use crate::position::Position;
...
+}
+
+impl Default for BoardState {
+    fn default() -> Self {
+        Self {
+            player: Color::default(),
+            board: Board::default(),
+            castling_rights: CastlingRights::default(),
+            en_passant_position: None,
+            white_king_position: Position::new(4, 0),
```
Comment (DanielAlbl) [inline chess/src/board_state.rs:29]: How about `CastlingRighst::WHITE_KING`?

File: chess/src/game.rs [hunk 2]
```diff
@@ -410,14 +417,52 @@ impl Game {
...
+            self.get_current_turn_index() / 2 + 1
         ));
         fen
     }
 
     pub fn get_current_move(&self) -> Option<Move> {
         self.history.get_current_move()
     }
+
+    pub fn get_highlighted_squares_info(&self) -> Vec<(Position, String)> {
```
Comment (DanielAlbl) [inline chess/src/game.rs:429]: Nice

File: chess/src/turn.rs [hunk 3]
```diff
@@ -7,21 +7,16 @@ pub struct Turn {
...
-        mv: Move,
-        piece_captured: bool,
-        king_is_checked: bool,
-    ) -> Self {
+    pub fn new(board_state: BoardState, mv: Move, piece_captured: bool) -> Self {
         Self {
             board_state,
             mv,
             piece_captured,
-            king_is_checked,
```
Comment (DanielAlbl) [inline chess/src/turn.rs:24]: Is there any downside to removing this?

File: chess/src/board_state.rs (hunk 1 above)
Comment (Dillon-Roller) [reply chess/src/board_state.rs:29]: Yeah

File: chess/src/turn.rs (hunk 3 above)
Comment (Dillon-Roller) [reply chess/src/turn.rs:24]: It's been replaced with the Game status, so it's more general now

File: chess/src/board_state.rs (hunk 1 above)
Comment (Dillon-Roller) [reply chess/src/board_state.rs]: I put it in `Position` instead

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**